openmc/tallies.py

```diff
@@ -3719,29 +3722,15 @@ def append(self, tally, merge=False):
                     # Replace tally2 with the merged tally
                     merged_tally = tally2.merge(tally)
                     self[i] = merged_tally
-                    merged = True
                     break
```

Comment:
Small thing, but cleans this up further. If this is `return` instead of `break` I think we can remove the double `else:` blocks below and leave one `super().append(tally)` to handle that case.